### PR TITLE
feat: duration argument is optional

### PR DESCRIPTION
**Pull request by monkindey**
fix https://github.com/ant-design/ant-design/issues/7857

**Any wrong, please let me know.**

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.
* [x] Update API docs for the component.
* [ ] Update/Add demo to demonstrate new feature.
* [x] Update TypeScript definition for the component.
* [x] Add unit tests for the feature.

### CODE REVIEW

File: components/message/index.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ type NoticeType = 'info' | 'success' | 'error' | 'warning' | 'loading';
 
 function notice(
   content: React.ReactNode,
-  duration: number = defaultDuration,
+  duration: (() => void) | number = defaultDuration,
```
Comment (infeng) [inline components/message/index.tsx]: Supply multiple function types for the same function as a list of overloads will be better.

Comment (monkindey) [reply components/message/index.tsx]: Thanks for your review, you mean ?

```js
duration: (() => void) | number = defaultDuration,
```

to 

```js
const fn = () => void;
...

duration: fn | number = defaultDuration,
```

Comment (abenhamdine) [reply components/message/index.tsx]: > Supply multiple function types for the same function as a list of overloads will be better.  

why is it better ?

Comment (infeng) [reply components/message/index.tsx]: Sorry, I got it wrong, you are right.

Comment (abenhamdine) [reply components/message/index.tsx]: No problem, just asking ;)

**Review by abenhamdine — APPROVED** [review]

**Comment by monkindey** [comment]
Thanks to be merged

**Pull request merged**